### ISSUE TITLE
Enum does not work as a response parameter

### ISSUE THREAD

**Issue by bombadile** [thecodingmachine/graphqlite#227]
Enum class:
```php
namespace App\Enum;

use MyCLabs\Enum\Enum;

class TaskStatus extends Enum
{
    public const STATUS_TO_DO = 1;
    public const STATUS_DOING = 2;
    public const STATUS_DONE = 3;
}
```
Entity:
```php 
/**
 * @Type()
 */
class Task
{   
    private int $status;

    /**
     * @Field()
     * @return TaskStatus
     */
    public function getStatusEnum(): TaskStatus
    {
        return new TaskStatus($this->status);
    }
}
```
GraphQL request:
```grahpql
query {
    getTask{status}
}
```

Result:
```JSON
{
  "errors": [
    {
      "type": "GraphQL\\Error\\InvariantViolation",
      "message": "Expected a value of type \"MyCLabsEnum___App__Enum__TaskStatus\" but received: 2",
      "file": "/app/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
      "line": 976,
      "trace": [
        {
          "file": "/app/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 976,
          "function": null,
          "class": "GraphQL\\Error\\InvariantViolation",
          "args": [
            "Expected a value of type \"MyCLabsEnum___App__Enum__TaskStatus\" but received: 2"
          ]
        },
        {
          "file": "[internal]",
          "line": 0,
          "function": "GraphQL\\Executor\\{closure}",
          "class": "GraphQL\\Executor\\ReferenceExecutor",
          "args": [
            [],
            "changeStatusTask"
          ]
        },
        {
          "file": "/app/vendor/webonyx/graphql-php/src/Type/Definition/EnumType.php",
          "line": 138,
          "function": null,
          "class": "GraphQL\\Error\\Error",
          "args": [
            "Cannot serialize value as enum: 2"
          ]
        },
        {
          "file": "/app/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 974,
          "function": "serialize",
          "class": "GraphQL\\Type\\Definition\\EnumType",
          "args": [
            2
          ]
        },
        {
          "file": "/app/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 846,
          "function": "completeLeafValue",
          "class": "GraphQL\\Executor\\ReferenceExecutor",
          "args": [
            "MyCLabsEnum___App__Enum__TaskStatus",
            2
          ]
        },
```

**Comment by moufmouf** [comment]
Hey @bombadile,

Thanks for opening this issue. I'll look into it.

**Comment by oojacoboo** [comment]
@moufmouf maybe I should open another issue for this, but I came to report another issue with Enum that seems related.  When you're outputting an Enum, the type is represented as the full `MyCLabsEnum___App__Enum__TaskStatus` namespace with the `\` replaced with `_`.

I looked into the `MyCLabsEnumTypeMapper` and see where this is happening here, but I'm not really sure on this logic and seems like it'd require a good deal of digging before I could figure it out.  I tried replacing the `name` in the `EnumType` object that's created to `(new \ReflectionClass($enumClass))->getShortName()` but webonyx complains that the type must be a unique name.  So, there is clearly more to it here.

Let me know if I should open a new issue ticket on this.
